pod.go

```diff
@@ -525,12 +528,100 @@ func (p *Pod) createSetStates() error {
 	return nil
 }
 
+type assetType string
+
+const (
+	kernelAsset assetType = "kernel"
+	imageAsset            = "image"
+)
+
+func validateAsset(path string, t assetType) bool {
+	// TODO Check for asset type
+	switch t {
+	case kernelAsset:
+		return true
+	case imageAsset:
+		return true
+	}
+
+	return true
+}
+
+func selectAssetFromAnnotation(podConfig *PodConfig, pathAnnotation, hashAnnotation string) (string, error) {
+	assetPath, ok := podConfig.Annotations[pathAnnotation]
+	if !ok {
+		return "", nil
+	}
+
+	assetHashString, ok := podConfig.Annotations[hashAnnotation]
+	if !ok {
+		return assetPath, nil
+	}
+
+	// We have a hash annotation, we need to verify it
+	assetBytes, err := ioutil.ReadFile(assetPath)
+	if err != nil {
+		return "", err
+	}
+
+	if len(assetHashString) != sha512.Size {
+		return "", fmt.Errorf("Invalid hash for %s", assetPath)
+	}
+
+	assetHashComputed := sha512.Sum512(assetBytes)
+
+	var assetHash [sha512.Size]byte
+	copy(assetHash[:], assetHashString)
+
+	if assetHashComputed != assetHash {
+		return "", fmt.Errorf("Invalid hash for %s", assetPath)
+	}
+
+	return assetPath, nil
+}
+
+func selectAssets(podConfig *PodConfig) error {
+	kernelPath, err := selectAssetFromAnnotation(podConfig, annotations.KernelPath, annotations.KernelHash)
+	if err != nil {
+		return err
+	}
+
+	imagePath, err := selectAssetFromAnnotation(podConfig, annotations.ImagePath, annotations.ImageHash)
+	if err != nil {
+		return err
+	}
+
+	// Do we want to update the hypervisor config only if both
+	// paths are not empty?
```

Comment:
I think we are good either way:
- require both paths to be filled out, to ensure the end user clearly states exactly what they need
or
- allow the end user to just over-ride one of the parts.

Maybe the former is better long term, as it could save somebody only setting one part, and then later down the line picking up an incompatible 'other part' during an upgrade, and getting hard to find problems then occurring...